File: FPG.Core/AlgorithmSet.cs
using System;

using FPG.Factories;
using FPG.Interfaces;

namespace FPG
{
    public class AlgorithmSet
    {
        public AlgorithmSet
            (String saltGenerationAlgorithmName,
            String saltingAlgorithmName,
            String hashingAlgorithmName,
            String croppingAlgorithmName,
            String symbolInsertionAlgorithmName)
        {
            this.SaltGeneration = SaltGenerationAlgorithmFactory.Create(saltGenerationAlgorithmName);
            this.Salting = SaltingAlgorithmFactory.Create(saltingAlgorithmName);
            this.Hashing = HashingAlgorithmFactory.Create(hashingAlgorithmName);
            this.Cropping = CroppingAlgorithmFactory.Create(croppingAlgorithmName);
            this.SymbolInsertion = SymbolInsertionAlgorithmFactory.Create(symbolInsertionAlgorithmName);
        }

        public ISaltGenerationAlgorithm SaltGeneration { get; }
        public ISaltingAlgorithm Salting { get; }
        public IHashingAlgorithm Hashing { get; }
        public ICroppingAlgorithm Cropping { get; }
        public ISymbolInsertionAlgorithm SymbolInsertion { get; }

        public static AlgorithmSet Create(String setName)
        {
            switch (setName.ToLower())
            {
                case "default":
                    return new AlgorithmSet("WP", "Additive", "SHA512", "Even", "NPicked");
                default:
                    throw new ArgumentException("Unknown algorithm set name");
            }
        }
    }
}

File: lib/core/AlgorithmSet.dart
import 'factories/CroppingAlgorithmFactory.dart';
import 'factories/HashingAlgorithmFactory.dart';
import 'factories/SaltGenerationAlgorithmFactory.dart';
import 'factories/SaltingAlgorithmFactory.dart';
import 'factories/SymbolInsertionAlgorithmFactory.dart';
import 'interfaces/ICroppingAlgorithm.dart';
import 'interfaces/IHashingAlgorithm.dart';
import 'interfaces/ISaltGenerationAlgorithm.dart';
import 'interfaces/ISaltingAlgorithm.dart';
import 'interfaces/ISymbolInsertionAlgorithm.dart';

class AlgorithmSet {
  AlgorithmSet(
      String saltGenerationAlgorithmName,
      String saltingAlgorithmName,
      String hashingAlgorithmName,
      String croppingAlgorithmName,
      String symbolInsertionAlgorithmName) {
    saltGeneration =
        SaltGenerationAlgorithmFactory.create(saltGenerationAlgorithmName);
    salting = SaltingAlgorithmFactory.create(saltingAlgorithmName);
    hashing = HashingAlgorithmFactory.create(hashingAlgorithmName);
    cropping = CroppingAlgorithmFactory.create(croppingAlgorithmName);
    symbolInsertion =
        SymbolInsertionAlgorithmFactory.create(symbolInsertionAlgorithmName);
  }

  late ISaltGenerationAlgorithm saltGeneration;
  late ISaltingAlgorithm salting;
  late IHashingAlgorithm hashing;
  late ICroppingAlgorithm cropping;
  late ISymbolInsertionAlgorithm symbolInsertion;

  static AlgorithmSet create(String setName) {
    switch (setName.toLowerCase()) {
      case "default":
        return AlgorithmSet("WP", "Additive", "SHA512", "Even", "NPicked");
      default:
        throw FormatException("Unknown algorithm set name");
    }
  }
}
